nofos/composer/tests/test_views.py

```diff
@@ -1157,3 +1174,202 @@ def test_404_document_mismatch(self):
         )
         resp = self.client.get(bad_url)
         self.assertEqual(resp.status_code, 404)
+
+
+class ComposerPreviewViewTests(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            email="test@example.com",
+            password="testpass123",
+            group="bloom",
+            force_password_reset=False,
+        )
+        self.client.login(email="test@example.com", password="testpass123")
+
+        self.guide = ContentGuide.objects.create(
+            title="Test Guide", opdiv="CDC", group="bloom", status="draft"
+        )
+        self.section = ContentGuideSection.objects.create(
+            content_guide=self.guide, order=1, name="Section 1", html_id="sec-1"
+        )
+        self.subsection = ContentGuideSubsection.objects.create(
+            section=self.section,
+            order=1,
+            name="Subsection 1",
+            tag="h3",
+            body="Test body",
+            enabled=True,
+        )
+
+        self.url = reverse("composer:composer_preview", kwargs={"pk": self.guide.pk})
+
+    def test_get_renders_preview_page(self):
+        resp = self.client.get(self.url)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.context["document"].pk, self.guide.pk)
+        self.assertTrue(resp.context["is_preview"])
+        self.assertIsNotNone(resp.context["sections"])
+
+    def test_anonymous_user_redirected(self):
+        self.client.logout()
+        resp = self.client.get(self.url)
+        self.assertIn(resp.status_code, [302, 403])
+
+    def test_post_exit_action_redirects_with_message(self):
+        resp = self.client.post(self.url, {"action": "exit"}, follow=False)
+
+        # Redirects to index
+        self.assertEqual(resp.status_code, 302)
+        self.assertEqual(resp.url, reverse("composer:composer_index"))
+
+        # Session variable set for success heading
+        follow_resp = self.client.get(resp["Location"])
+        self.assertEqual(follow_resp.context["success_heading"], "Content guide successfully saved")
+
+        # Success message was added
+        msgs = list(get_messages(resp.wsgi_request))
+        self.assertTrue(any("You saved:" in str(m) for m in msgs))
+        self.assertTrue(any(self.guide.title in str(m) for m in msgs))
```

Comment:
Maybe it is a personal preference, but I prefer to just search for the exact string of the message because these get hard to read. 

here is an example from my PR:

```
        messages = [m.message for m in get_messages(response.wsgi_request)]
        self.assertTrue(
            any("Draft NOFO “First Draft” created successfully." in m for m in messages)
        )
```

The downside is that they are more brittle (changing anything in the message can break the test), but arguably given how short our messages are, we are already close with these partial matches.